Classes/Controller/Cart/OrderController.php

```diff
@@ -83,7 +85,37 @@ public function createAction(
             return $this->redirect('show', 'Cart\Cart');
         }
 
-        $this->eventDispatcher->dispatch(new ProcessOrderCheckStockEvent($this->cart));
+        $processOrderCheckStockEvent = new ProcessOrderCheckStockEvent($this->cart);
+        $this->eventDispatcher->dispatch($processOrderCheckStockEvent);
+
+        if (!$processOrderCheckStockEvent->isEveryProductAvailable()) {
+            $errors = $processOrderCheckStockEvent->getInsufficientStockMessages();
+
+            $messageBody = '';
+            $messageTitle = '';
+            $severity = ContextualFeedbackSeverity::OK;
+
+            foreach ($errors as $error) {
+                if (!($error instanceof AbstractMessage)) {
+                    continue;
+                }
+                $message = $error->jsonSerialize();
+                if ($message['severity'] >= (int)$severity) {
+                    $severity = $message['severity'];
+                    $messageBody = $message['message'];
+                    $messageTitle = $message['title'];
+                }
+
+                $this->addFlashMessage(
+                    $messageBody,
+                    $messageTitle,
+                    $severity,
+                    true
+                );
+            }
```

Comment:
If all insufficientStockMessages should be displayed, the code can reduced to:

```suggestion
            foreach ($errors as $error) {
                $error->setStoreInSession(true);
                $this->getFlashMessageQueue()->enqueue($error);
            }
```

_(not tested)_

Reply:
Done.